### ISSUE TITLE
Disabled toolbar icons are no longer dimmed in 1.67

### ISSUE THREAD

**Issue by gareth-rees** [microsoft/vscode#149075]
### Summary

In Visual Studio Code 1.65 and earlier, icons added by extensions to the debug toolbar were dimmed when disabled:

![Screenshot from 2022-05-09 10-57-26](https://user-images.githubusercontent.com/922721/167389277-52f6b7ce-ce4d-4d86-a4ca-3c1d4654ca10.png)

But in Visual Studio Code 1.67.0, icons added by extensions to the debug toolbar are no longer dimmed when disabled:

![Screenshot from 2022-05-06 15-11-10](https://user-images.githubusercontent.com/922721/167389505-763cc779-b7d2-4688-b838-360fb7b9336a.png)

### Details

I am working on an extension which adds icons to the debug toolbar, using the [`"commands"` contribution point](https://code.visualstudio.com/api/references/contribution-points#contributes.commands). The icons are disabled while the program is running, and enabled when it is stopped. A typical icon definition looks like this:
```
"commands": [
    {
        "command": "udb.reverseStepOver",
        "enablement": "debugState == 'stopped' && udbActive",
        "title": "Reverse Step Over",
        "icon": {
            "dark": "media/stepover-reverse-tb-dark.svg",
            "light": "media/stepover-reverse-tb-light.svg"
        }
    },
],
```
with a corresponding definition in the [`"menus"` contribution point](https://code.visualstudio.com/api/references/contribution-points#contributes.menus):
```
"menus": {
    "debug/toolBar": [
        {
            "command": "udb.reverseStepOver",
            "when": "debugType == 'cppdbg' && udbActive",
            "group": "navigation@6"
        },
    ]
},
```

### Discussion

It seems that Visual Studio Code 1.67 no longer provides a way for extensions to vary the colour of toolbar icons when they are disabled.

Looking at the Visual Studio Code source, it seems that the built-in toolbar icons are created like this:

https://github.com/microsoft/vscode/blob/3d7135a49f5afbadc6849cce343666eaa446ea66/src/vs/workbench/contrib/debug/browser/debugIcons.ts#L65

and then styled like this:

https://github.com/microsoft/vscode/blob/3d7135a49f5afbadc6849cce343666eaa446ea66/src/vs/workbench/contrib/debug/browser/debugColors.ts#L104-L109

https://github.com/microsoft/vscode/blob/3d7135a49f5afbadc6849cce343666eaa446ea66/src/vs/workbench/contrib/debug/browser/debugColors.ts#L340-L343

but I cannot see any way for extensions to access this functionality.

**Comment by roblourens** [comment]
I see this, hey @daviddossett this is from eec956770ef6bb6fe904460a912233b93accd609. Seems like this will work for codicons but not svgs. Any idea how to deal with that? 

We could add a new class to svg icons so we can keep the opacity rule, if that helps. This probably applies to all toolbars, besides the debug one.

**Comment by gareth-rees** [comment]
In the short term, restoring the opacity rule for SVG icons makes the most sense.

But in the medium term, it would be nice if extensions had a way to access codicon-like behaviour, perhaps via the [`"icons"` contributions point](https://code.visualstudio.com/api/references/contribution-points#contributes.icons). At the moment there is no way to for extensions to vary the colour for icons defined via this contribution point, but built-in icons use `fill="currentColor"` in their SVG, with the colour being specified via CSS.

**Comment by roblourens** [comment]
Let's take this as a candidate. Could you also prepare a PR against the release branch?

**Comment by gareth-rees** [comment]
@daviddossett Thanks for the super-quick fix! Look forward to trying it out in 1.67.2 (I assume).

> can you clarify what you're hoping to achieve re: codicon-like behavior? Are you hoping to use theme tokens in your contributed SVGs?

Concretely, what I would like to achieve is to be able to put icons in the debug toolbar that look like this when the program is stopped:

![Screenshot from 2022-05-10 18-37-34](https://user-images.githubusercontent.com/922721/167691441-1b185fa8-7583-4290-918f-d7900ab4fc28.png)

and like this when the program is running (where grayed-out icons represent commands that are temporarily disabled):

![Screenshot from 2022-05-10 18-48-48](https://user-images.githubusercontent.com/922721/167691437-66364f5a-46f6-49ee-bc35-ea01d01a8c54.png)

In these screenshots, six of the commands are using bulit-in icons (Continue/Pause, Step Over, Step Into, Step Out, Restart and Stop), and the other eleven commands are using icons defined by the extension. I would like the two sets of icons to match in style if possible. If I can choose colours from the user's theme, that would be perfect, but I don't mind exactly how the functionality is provided: you know better than me what implementation would make the most sense.
